src/common/types/tasks.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+import type { FileMini, FolderMini, ISODate, UserMini } from './core';
+
+type ID = string;
+
+type TaskCollabStatus = 'APPROVED' | 'REJECTED' | 'COMPLETED' | 'NOT_STARTED';
+
+type TaskStatus = TaskCollabStatus | 'IN_PROGRESS';
+
+type TaskMini = {|
+    created_at: ISODate,
+    id: ID,
+    modified_at: ISODate,
+    status: TaskStatus,
+    type: 'task',
+|};
+
+type TaskCollab<R> = {|
+    completed_at?: ?ISODate,
+    id: ID,
+    role: R,
+    status: TaskCollabStatus,
+    target: UserMini,
+    task?: TaskMini,
+    type: 'task_collaborator',
+|};
+
+type TaskCollabCreator = TaskCollab<'CREATOR'>;
+
+type TaskCollabAssignee = {|
+    ...TaskCollab<'ASSIGNEE'>,
+    permissions: {|
+        can_delete: boolean,
+        can_update: boolean,
+    |},
+|};
+
+type TaskLink = {|
+    description?: string,
+    id: ID,
+    permissions: {|
+        can_delete: boolean,
+        can_update: boolean,
+    |},
+    target?: ?FileMini | ?FolderMini | ?UserMini,
+    task?: TaskMini,
+    type: 'task_link',
+|};
+
+type MarkerPaginatedCollection<T> = {
```

Comment:
Seems like this should probably still live in an even more common location, since quite a few V2 endpoints response with this format.